tasks/main.yml

```diff
@@ -340,7 +340,7 @@
         __mssql_conf_setting_value: "{{ mssql_tune_for_fua_storage }}"
 
 - name: Configure MSSQL to encrypt connections
-  when: mssql_tls.enable
+  when: mssql_tls.enable | bool
```

Comment:
I think this is ok - because this is an operation (the `.` is a dereference operation) it will be evaluated correctly - but it is ok to leave this with the `| bool` for now

Reply:
I am confused about this bareword issue. I expect the value of `mssql_tls.enable` to be a boolean, do I need to add `| bool` in that case? I'd rather wish the role to fail if the user provides NOT a boolean value. I type this in the readme. If I only expect boolean, a string `"true"` (not a bool `true`) won't work, but I think it is fine.

Reply:
> I am confused about this bareword issue.

It happens when you use a boolean valued variable in a `when` or `assert` `that` by itself - no operations, no filters - that is, the variable is a bare word:
```
when: somevalue
```
I can't remember the reason that Ansible/yaml/jinja cannot parse this correctly as a boolean value.
If the word is not "bare" - if you use another conditional like `not`, or use a dereference operator, or a filter, then Ansible/yaml/jinja are able to parse it correctly.  So all of these are ok - they are not "bare":
```
when: mssql_tls.enable  # dereference operator

when: not somevalue  # uses "not"

when: somevalue | bool  # uses a filter
```

> I expect the value of `mssql_tls.enable` to be a boolean, do I need to add `| bool` in that case?

The only other case where you would use `| bool` is when the value of `mssql_tls.enable` is a string that you must convert to a boolean value.

> I'd rather wish the role to fail if the user provides NOT a boolean value. I type this in the readme. If I only expect boolean, a string `"true"` (not a bool `true`) won't work, but I think it is fine.

Ok - if `mssql_tls.enable` can be a string `"true"` or `"false"`, then you should use `| bool` here.

Reply:
Got it, thank you for the explanation. I think it is fine to keep `| bool` in this case, and in all other cases when checking booleans for the sake of consistency. `| bool` doesn't break anything.